src/github.com/getlantern/lantern-mobile/app/src/main/java/org/getlantern/lantern/activity/LanternMainActivity.java

```diff
@@ -104,57 +182,289 @@ protected void onCreate(Bundle savedInstanceState) {
+        });
+
+        menuMap.put(resources.getString(R.string.newsfeed_option), new Command() {
+            public void runCommand() {
+                updateFeedview(listAdapter, mNavItems, resources, 3, true);
+            }
+        });
+
+        menuMap.put(resources.getString(R.string.desktop_option), new Command() { 
+            public void runCommand() {
+                startActivity(new Intent(activity, DesktopActivity_.class));
+            } 
+        });
+
+        menuMap.put(resources.getString(R.string.share_option), new Command() { 
+            public void runCommand() {
+                final Shareable shareable = new Shareable(activity);
+                if (shareable != null) {
+                    shareable.showOption();
+                }
+            }
+        });   
+
+        // Populate the Navigtion Drawer with options
+        drawerList.setAdapter(listAdapter);
+
+        // remove ListView border
+        drawerList.setDivider(null);
+
+        // Drawer Item click listeners
+        drawerList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+
+                drawerList.setItemChecked(position, true);
+                if (position >= 0 && position < mNavItems.size()) {
+                    NavItem item = mNavItems.get(position);
+                    if (item != null) {
+                        String title = item.getTitle();
+                        Log.d(TAG, "Menu option " + title + " selected");
+                        menuMap.get(title).runCommand();
+                    }
+                }
+
+                // Close the drawer
+                drawerLayout.closeDrawer(drawerPane);
+            }
+        });
+
+        mDrawerToggle = new ActionBarDrawerToggle(this, drawerLayout, R.string.drawer_open, R.string.drawer_close) {
+            @Override
+            public void onDrawerOpened(View drawerView) {
+                super.onDrawerOpened(drawerView);
+                invalidateOptionsMenu();
+            }
+
+            @Override
+            public void onDrawerClosed(View drawerView) {
+                super.onDrawerClosed(drawerView);
+                Log.d(TAG, "onDrawerClosed: " + getTitle());
+                invalidateOptionsMenu();
+            }
+        };
+
+        drawerLayout.setDrawerListener(mDrawerToggle);
+    }
+
+    @Click(R.id.menuIcon)
+    void menuButtonClicked() {
+        drawerLayout.openDrawer(Gravity.START);
+    }
+
+    // showFeedview optionally fetches the feed depending on the
+    // user's preference and updates the position of the on/off switch
+    private void showFeedview() {
+        RelativeLayout.LayoutParams lp = (RelativeLayout.LayoutParams) powerLantern.getLayoutParams();
+
+        if (session.showFeed()) {
+            feedView.setVisibility(View.VISIBLE);
+            lp.removeRule(RelativeLayout.CENTER_VERTICAL);
```

Comment:
Apparently, you can call ``addRule(rule, 0)`` to achieve the same effect as ``removeRule`` in earlier versions. I added a check to pick the right one to call depending on the API level.